src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1634,6 +1638,46 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::check_service_node_time()
+  {
+    crypto::public_key pubkey = m_service_node_list.get_random_pubkey();
+    crypto::x25519_public_key x_pkey{0};
+    constexpr std::array<uint16_t, 3> MIN_TIMESTAMP_VERSION{8,1,5};
+    std::array<uint16_t,3> proofversion;
+    m_service_node_list.access_proof(pubkey, [&](auto &proof) { 
+      x_pkey = proof.pubkey_x25519; 
+      proofversion = proof.version;
+    });
+    if (proofversion >= MIN_TIMESTAMP_VERSION && x_pkey) {
+      m_lmq->request(
+        std::to_string(x_pkey),
+        "quorum.timestamp",
+        [&, pubkey](bool success, std::vector<std::string> data) {
```

Comment:
LMQ request callbacks get invoked at a later time, i.e. after this function has returned, which means you want to be extra careful about not keep references to local variables, which you could end up doing accidentally with the `&` default.  I don't think there's anything problematic here right now, but it guards against accidentally adding a reference in the future.

But in this case you can slightly clean it up by capturing just: `[this, pubkey]` instead of `[&, pubkey]` -- that should give you access to everything you need here without any other changes.